pkg/authentication/authentication.go

```diff
@@ -33,3 +34,35 @@ func Enable(ctx context.Context, auth *automationconfig.Auth, secretGetUpdateCre
 	}
 	return nil
 }
+
+func AddRemovedUsers(auth *automationconfig.Auth, mdb mdbv1.MongoDBCommunity, lastAppliedSpec *mdbv1.MongoDBCommunitySpec) {
+	deletedUsers := getDeletedUsers(mdb.Spec, lastAppliedSpec)
+
+	auth.UsersDeleted = append(auth.UsersDeleted, deletedUsers...)
+}
+
+func getDeletedUsers(currentMDB mdbv1.MongoDBCommunitySpec, lastAppliedMDBSpec *mdbv1.MongoDBCommunitySpec) []automationconfig.DeletedUser {
```

Comment:
I think a name more like `getRemovedUsersFromSpec` is more fitting. But I will leave that to you